Cartesian_kernel/include/CGAL/Cartesian/function_objects.h

```diff
@@ -3289,6 +3291,14 @@ namespace CartesianKernelFunctors {
       line_project_pointC2(l.a(), l.b(), l.c(), p.x(), p.y(), x, y);
       return construct_point_2(x, y);
     }
+
+    Point_2
+    operator()(const Segment_2& s, const Point_2& p) const
```

Comment:
Adding new overloads in the models of `Kernel` without any change of `Kernel` looks strange.

Reply:
I just add it to the documentation for segment and triangle.   As in the 3D version it is a projection to the closest point.